src/main/java/com/novemberain/quartz/mongodb/dao/TriggerDao.java

```diff
@@ -70,12 +65,12 @@ public boolean exists(Bson filter) {
         return triggerCollection.count(filter) > 0;
     }
 
-    public FindIterable<Document> findEligibleToRun(Date noLaterThanDate) {
+    public FindIterable<Document> findEligibleToRun(Date noLaterThanDate, Integer limit) {
         Bson query = createNextTriggerQuery(noLaterThanDate);
-        if (log.isInfoEnabled()) {
-            log.info("Found {} triggers which are eligible to be run.", getCount(query));
-        }
-        return triggerCollection.find(query).sort(ascending(Constants.TRIGGER_NEXT_FIRE_TIME));
+
+        return triggerCollection.find(query)
+                .sort(ascending(Constants.TRIGGER_NEXT_FIRE_TIME))
+                .limit(limit);
```

Comment:
This change is fine for me, however limit cannot be null therefore should be "int".
Please add tests for this method in TriggerDaoTest.